File: b2b.py
import africastalking

africastalking.initialize(username="", api_key="")

payment = africastalking.Payment

product_name = ""

business = {
    "provider": provider,
    # Set the destination channel and destination account
	"destinationChannel": "partnerBusinessChannel",
    "destinationAccount": "partnerBusinessChannelAccount",
	# Set the transfer type
    "transferType": "BusinessToBusinessTransfer",
    # Set The 3-Letter ISO currency code and the checkout amount
    "currencyCode": currency_code,
    "amount": amount,
    # Set any metadata that you would like to send along with this request.
    # This metadata will be included when we send back the final payment notification
    "metadata": {
        "shopId" => "1234",
        "itemId" => "abcde"
    }

}

try:
    res = payment.mobile_b2b(product_name, business)
    print(res)
except Exception as e:
    print(f"Houston we have a problem: {e}")

File: b2b.js
//Set up app credentials
const credentials = {
    apiKey: "YOUR_API_KEY_GOES_HERE",
    username: "YOUR_USERNAME_GOES_HERE"
}

//Initialize the AT SDK
const AfricasTalking = require("africastalking")(credentials)

//Call the Payments service
const payments = AfricasTalking.PAYMENTS

const initiateMobileB2B = async () => {
    
    
    const options = {
        //Set up the name of the AT Payment Product name
        productName: "PRODUCT_NAME",
        //Set up the payments provider
        provider: "PAYMENT_PROVIDER",
        //Set the destination channel and account
        destinationChannel: "partnerBusinessChannel",
        destinationAccount: "partnerAccountName",
        //Set the transer type
        transferType: "",
        //Set the 3-letter ISO currency code and amount
        currencyCode: "",
        amount: "",
        //Optional metadata 
        metadata: {
            foo: "bar",
            bar: "foo"
        }
    }

    //That's it. We'll take care of the rest
    try {
        const result = await payments.mobileB2B(options)
        console.log(result)
    } catch (err) {
        console.log(err)
    }
}